src/SFA.DAS.PR.Jobs/Functions/Notifications/SendNotificationsFunction.cs

```diff
@@ -64,13 +70,29 @@ public async Task Run([TimerTrigger("%SendNotificationsFunctionSchedule%", RunOn
         _logger.LogInformation("{FunctionName} - Processed {ProcessedCount} notifications.", nameof(SendNotificationsFunction), processedCount);
     }
 
-    private async Task<int> TryProcessProviderNotification(Notification notification, CancellationToken cancellationToken)
+    private async Task<int> TryProcessProviderNotification(Notification notification, FunctionContext executionContext, CancellationToken cancellationToken)
     {
         try
         {
-            ProviderEmailRequest providerEmailRequest = await CreateProviderEmailRequest(notification, cancellationToken);
-            await _pasAccountApiClient.SendEmailToAllProviderRecipients(notification.Ukprn!.Value, providerEmailRequest, cancellationToken);
-            notification.SentTime = DateTime.UtcNow;
+            switch(notification.NotificationType)
+            {
+                case nameof(NotificationType.Provider):
+                    {
+                        ProviderEmailRequest providerEmailRequest = await CreateProviderEmailRequest(notification, cancellationToken);
+
+                        await _pasAccountApiClient.SendEmailToAllProviderRecipients(notification.Ukprn!.Value, providerEmailRequest, cancellationToken);
+                    }
+                    break;
+                case nameof(NotificationType.Employer):
+                    {
+                        SendEmailCommand employerEmailRequest = await CreateEmployerEmailRequest(notification, cancellationToken);
+
+                        await _functionEndpoint.Send(employerEmailRequest, executionContext, cancellationToken);
+                    }
+                    break;
+            }
+
+            await UpdateNotification(notification, cancellationToken);
```

Comment:
We need to commit each notification as it is sent out. Currently if any one fails all of the notifications will be retried, which will mean that we will send out duplicate notifications.